lib/trace_mgr.js

```diff
@@ -0,0 +1,611 @@
+        }
+    }
+
+    //update the bin stats as needed
+    updateEmitInfo(entry, resolvedPath);
+
+    return buildTraceResult('success', action);
+}
+
+//create a directory for emitting the trace (if possible) and return it
+function createTraceLogTarget(tracename) {
+    var traceRootDir = emitOptions.localTraceDirectory;
+    if (traceRootDir === undefined) {
+        traceRootDir = path.dirname(process.mainModule.filename);
+    }
+    resolvedTracePath = path.resolve(traceRootDir, '_diagnosticTraces', tracename);
+
+    //ensure directory exists and is empty...
+    var ok = ensureTraceTarget(resolvedTracePath);
+    if (!ok) {
+        return undefined;
+    }
+
+    return resolvedTracePath;
+}
+
+//ensure directory exists and is empty...
+function ensureTraceTarget(pth) {
+    //I don't like this and don't want it to be happening here so I am going to bail
+    if (!path.isAbsolute(pth)) {
+        return false;
+    }
+
+    var okdir = createTargetDirectory(pth);
+    if (!okdir) {
+        return false;
+    }
+
+    return deleteTargetDirectoryContents(pth);
+}
+
+function createTargetDirectory(pth) {
+    //see if it just exists and, if so, just return true
+    try {
+        fs.accessSync(pth, fs.constants.R_OK | fs.constants.W_OK | fs.constants.X_OK);
+        return true;
+    }
+    catch (ei) { ; }
+
+    //walk up the directory to see where the first valid part of the path is
+    var prefixPath = pth;
+    var suffixPaths = [];
+    var baseFound = false;
+    do {
+        //check for bad prefix 
+        if (prefixPath === path.dirname(prefixPath)) {
+            process.stderr.write(`Failed creating trace directory (bad prefix): ${pth} -> ${prefixPath}\n`);
+            return false;
+        }
+
+        suffixPaths.push(path.basename(prefixPath)); //reverse order 
+        prefixPath = path.dirname(prefixPath);
+
+        try {
+            fs.accessSync(prefixPath, fs.constants.R_OK | fs.constants.W_OK | fs.constants.X_OK);
+            baseFound = true;
+        }
+        catch (ei) { ; }
+    } while (!baseFound);
+
+    //now extend the prefix with all the suffix parts
+    while (suffixPaths.length > 0) {
+        try {
+            prefixPath = path.resolve(prefixPath, suffixPaths.pop());
+            fs.mkdirSync(prefixPath);
+        }
+        catch (ec) {
+            process.stderr.write(`Failed creating trace directory (failed mkdir): ${ec}\n`);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+function deleteTargetDirectoryContents(pth) {
+    try {
+        var items = fs.readdirSync(pth);
+        for (var i = 0; i < items.length; i++) {
+            var fpath = path.resolve(pth, items[i]);
+            var stats = fs.lstatSync(fpath);
+            if (stats.isFile()) {
+                fs.unlinkSync(fpath);
+            }
+            else if (stats.isDirectory()) {
+                deleteTargetDirectoryContents(fpath);
```

Comment:
This can return false since it wraps its code with an exception- probably want to just return if this fails rather than executing rmdirSync and having it throw again